utils.js

```diff
@@ -202,6 +225,14 @@ var verifyDependency = function(recordarray, record) {
               return
             }
           }
+          if (n.readfrom === '$') {
+            //deep copy
+            tempvalue = JSON.parse(JSON.stringify(recordarray[n.record].info.data))
+            return
+          }
+          if(!!recordarray['state'] && !!recordarray['state'].barVariables){
+            n.readfrom = mustache.render(n.readfrom, recordarray['state'].barVariables)
```

Comment:
We need to make it generic so that anyone can use any record (already resolved) and use that info to render.